src/db.c

```diff
@@ -1325,6 +1332,39 @@ void scanDatabaseForReadyLists(redisDb *db) {
     dictReleaseIterator(di);
 }
 
+/* Since we are unblocking XREADGROUP clients in the event the
+ * key was deleted/overwritten we must do the same in case the
+ * database was flushed/swapped. */
+void scanDatabaseForDeletedStreams(redisDb *emptied, redisDb *replaced_with) {
+    /* Optimization: If no clients are in type BLOCKED_STREAM,
+     * we can skip this loop. */
+    if (!server.blocked_clients_by_type[BLOCKED_STREAM]) return;
+
+    dictEntry *de;
+    dictIterator *di = dictGetSafeIterator(emptied->blocking_keys);
+    while((de = dictNext(di)) != NULL) {
+        robj *key = dictGetKey(de);
+        int was_stream = 0, is_stream = 0;
+
+        dictEntry *kde = dictFind(emptied->dict, key->ptr);
+        if (kde) {
+            robj *value = dictGetVal(kde);
+            was_stream = value->type == OBJ_STREAM;
+        }
+        if (replaced_with) {
```

Comment:
I think we don't need to check if the deleted stream is replaced with no-stream object:

1. if the new object is not stream, it means the original key is overwritten, should awaken the blocked client.
2. if the new object is stream, it may not have the specific group, should also awaken the blocked client to check if it needs to reply an `-NOGROUP` error.

Reply:
1. not necessarily. if the new type of the key isn't list/zset/stream/module, no one will call `signalKeyAsReady` and the client will remain blocked.
2. this statement is correct (but it's not related to this function, which only checks if the new key is not a stream)

Reply:
I mean if the current stream is replaced with another db's stream, and the new stream doesn't have the original stream's group, we should also wake up the client in case it blocked forever.

Reply:
because the `SWAPDB` command on a stream key is like a transaction:
```
MULTI
DEL stream
XADD stream ...
EXEC
```

Reply:
OK, I forgot you call `scanDatabaseForReadyKeys` after db swapped, I get your point now.